File: src/dlp/ast/expressions/ArrayAccess.ts

import {Expression} from "./Expression";
import {AbstractLocatable} from "../AbstractLocatable";

export class ArrayAccess extends AbstractLocatable implements Expression {

    access: Expression;
    value: Expression;

    constructor( access: Expression, value: Expression, line: number, column: number){
        super(line, column);
        this.access = access;
        this.value = value;
    }
}

File: src/ast/expressions/ArrayAccess.java
package ast.expressions;

import ast.AbstractLocatable;
import visitors.Visitor;

public class ArrayAccess extends AbstractExpression {

    private Expression access;
    private Expression value;

    public ArrayAccess(Expression access, Expression value, int line, int column) {
        super(line, column);
        this.access = access;
        this.value = value;
    }

    public <PT,RT> RT accept(Visitor<PT,RT> v, PT param) {
        return v.visit(this, param);
    }

    public Expression getAccess() {
        return access;
    }

    public Expression getValue() {
        return value;
    }
}
